paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -215,5 +232,84 @@ def sync_secrets(
     return True
 
 
+def sync_boto_secrets(
+    kube_client: KubeClient,
+    cluster: str,
+    service: str,
+    secret_provider_name: str,
+    vault_cluster_config: Mapping[str, str],
+    soa_dir: str,
+) -> bool:
+    # Update boto key secrets
+    config_loader = PaastaServiceConfigLoader(service=service, soa_dir=soa_dir)
+    for instance_config in config_loader.instance_configs(
+        cluster=cluster, instance_type_class=KubernetesDeploymentConfig
+    ):
+        instance = instance_config.instance
+        boto_keys = instance_config.config_dict.get("boto_keys", [])
+        if not boto_keys:
+            continue
+        boto_keys.sort()
+        secret_data = {}
+        for key in boto_keys:
+            for filetype in ["sh", "yaml", "json", "cfg"]:
+                this_key = key + "." + filetype
+                sanitised_key = this_key.replace(".", "-").replace("_", "--")
+                try:
+                    with open(f"/etc/boto_cfg_private/{this_key}") as f:
+                        secret_data[sanitised_key] = base64.b64encode(
+                            f.read().encode("utf-8")
+                        ).decode("utf-8")
+                except IOError:
+                    log.warning(
+                        f"Boto key {this_key} required for {service} could not be found."
+                    )
+        if not secret_data:
+            continue
+        secret = f"paasta-boto-key-{get_kubernetes_app_name(service, instance)}"
+        hashable_data = "".join([secret_data.get(key, "") for key in secret_data])
```

Comment:
we should be able to use `secret_data[key]` here, right? I'm just trying to figure out in what cases we'd get to this line and potentially have a missing key in this dict

Reply:
Yeah, it's literally got `for key in secret_data` ...